core/services/ocr2/plugins/ocr2vrf/fixed_blockhash_provider.go

```diff
@@ -0,0 +1,109 @@
+package ocr2vrf
+
+import (
+	"context"
+	"math/big"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/smartcontractkit/ocr2vrf/types"
+
+	"github.com/smartcontractkit/chainlink/core/chains/evm/client"
+)
+
+// Implementation of Blockhashes interface that returns blockhashes with fixed-sized windows relative to current block height
+type FixedBlockhashProvider struct {
+	client client.Client
+	// end block = current head - waitBlocks
+	waitBlocks uint
+	// start block = current head - lookbackBlocks
+	lookbackBlocks uint
+	// number of blocks to query in a batch
+	batchSize uint
+}
+
+type Block struct {
+	Number    int64
+	Hash      common.Hash
+	Timestamp time.Time
+}
+
+var _ types.Blockhashes = (*FixedBlockhashProvider)(nil)
+
+func NewFixedBlockhashProvider(client client.Client, waitBlocks, lookbackBlocks, batchSize uint) *FixedBlockhashProvider {
+	return &FixedBlockhashProvider{
+		client,
+		waitBlocks,
+		lookbackBlocks,
+		batchSize,
+	}
+}
+
+func (b *FixedBlockhashProvider) OnchainVerifiableBlocks(
+	ctx context.Context,
+) (startHeight uint64, hashes []common.Hash, err error) {
+	height, err := b.CurrentHeight(ctx)
+	if err != nil {
+		return 0, nil, err
+	}
+	fromBlock := height - uint64(b.lookbackBlocks)
+	toBlock := height - uint64(b.waitBlocks)
+	reqs := []rpc.BatchElem{}
```

Comment:
Style: prefer `var reqs []rpc.BatchElem` to `reqs := []rpc.BatchElem{}` (see https://github.com/uber-go/guide/blob/master/style.md#nil-is-a-valid-slice)

Reply:
that's a nice guide. I'll go through this soon!